# 01 Importing Libraries

In [2]:
import pandas as pd
import numpy as ny
import os

# 02 Importing US birth data set

In [3]:
df_US = pd.read_csv(r'C:\Users\eniad\Downloads\Achievement 6\02 Data\Original Data\US birth numbers.csv', index_col=False)

In [4]:
df_US.head()

State State Abbreviation  Year Gender  \
0  Alabama                 AL  2016      F   
1  Alabama                 AL  2016      F   
2  Alabama                 AL  2016      F   
3  Alabama                 AL  2016      F   
4  Alabama                 AL  2016      F   

                Education Level of Mother  Education Level Code  \
0                       8th grade or less                     1   
1  9th through 12th grade with no diploma                     2   
2   High school graduate or GED completed                     3   
3   Some college credit, but not a degree                     4   
4               Associate degree (AA, AS)                     5   

   Number of Births  Average Age of Mother (years)  Average Birth Weight (g)  
0              1052                           27.8                    3116.9  
1              3436                           24.1                    3040.0  
2              8777                           25.4                    3080.0  
3              6453                           26.7                    3121.9  
4              2227                           28.9                    3174.3

In [5]:
df_US.head(100)

State State Abbreviation  Year Gender  \
0   Alabama                 AL  2016      F   
1   Alabama                 AL  2016      F   
2   Alabama                 AL  2016      F   
3   Alabama                 AL  2016      F   
4   Alabama                 AL  2016      F   
..      ...                ...   ...    ...   
95  Alabama                 AL  2021      F   
96  Alabama                 AL  2021      F   
97  Alabama                 AL  2021      F   
98  Alabama                 AL  2021      F   
99  Alabama                 AL  2021      M   

                            Education Level of Mother  Education Level Code  \
0                                   8th grade or less                     1   
1              9th through 12th grade with no diploma                     2   
2               High school graduate or GED completed                     3   
3               Some college credit, but not a degree                     4   
4                           Associate degree (AA, AS)                     5   
..                                                ...                   ...   
95                     Bachelor's degree (BA, AB, BS)                     6   
96      Master's degree (MA, MS, MEng, MEd, MSW, MBA)                     7   
97  Doctorate (PhD, EdD) or Professional Degree (M...                     8   
98                              Unknown or Not Stated                    -9   
99                                  8th grade or less                     1   

    Number of Births  Average Age of Mother (years)  Average Birth Weight (g)  
0               1052                           27.8                    3116.9  
1               3436                           24.1                    3040.0  
2               8777                           25.4                    3080.0  
3               6453                           26.7                    3121.9  
4               2227                           28.9                    3174.3  
..               ...                            ...                       ...  
95              4480                           30.5                    3217.7  
96              2173                           31.9                    3212.8  
97               577                           33.0                    3264.8  
98                32                           27.8                    3056.9  
99              1203                           27.6                    3228.6  

[100 rows x 9 columns]

In [6]:
df_US.shape

(5496, 9)

# 03 CREATING DESCRIPTIVE STATISTICS

In [7]:
# Checking consistency on df_US
df_US.describe()

Year  Education Level Code  Number of Births  \
count  5496.000000           5496.000000       5496.000000   
mean   2018.500728              3.026201       4115.443959   
std       1.707554              4.733406       6687.036797   
min    2016.000000             -9.000000         10.000000   
25%    2017.000000              2.000000        559.000000   
50%    2019.000000              4.000000       1692.000000   
75%    2020.000000              6.000000       5140.000000   
max    2021.000000              8.000000      59967.000000   

       Average Age of Mother (years)  Average Birth Weight (g)  
count                    5496.000000               5496.000000  
mean                       29.552274               3250.887627  
std                         2.779735                114.456090  
min                        23.100000               2451.900000  
25%                        27.500000               3182.200000  
50%                        29.600000               3256.000000  
75%                        31.800000               3330.825000  
max                        35.500000               3585.700000

From the above table, the data looks consistent however, there may be need to investigate 'Number of births' the minimum value is 10 while the maximum is 59967 and thats looks like an outliers.

# 04 CHECKING for MIXED-TYPE COLUMN

In [8]:
# create a dataframe
df_test = pd.DataFrame()

In [9]:
# Create a mixed-type column
df_test['mix'] = ['a', 'b', 1, True]

In [10]:
df_test.head()

mix
0     a
1     b
2     1
3  True

In [11]:
#Creating mixed type column
for col in df_test.columns.tolist():
  weird = (df_test[[col]].applymap(type) != df_test[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_test[weird]) > 0:
    print (col)

mix


Having checked the data, there is no mixed-type data from the data 

# 05 CHECKING FOR MISSING VALUES

In [12]:
# Checking for missing values
df_US.isnull().sum()

State                            0
State Abbreviation               0
Year                             0
Gender                           0
Education Level of Mother        0
Education Level Code             0
Number of Births                 0
Average Age of Mother (years)    0
Average Birth Weight (g)         0
dtype: int64

The above check shows that there is no missing values from the data

# 06 CHECKING FOR DUPLICATES

In [13]:
df_dups = df_US[df_US.duplicated()]

In [17]:
df_dups

Empty DataFrame
Columns: [State, State Abbreviation, Year, Gender, Education Level of Mother, Education Level Code, Number of Births, Average Age of Mother (years), Average Birth Weight (g)]
Index: []

The table shows no duplicates in the data set

In [20]:
df_dups.shape

(0, 9)

# 07 EXPORTING df_US 

In [21]:
df_US.to_csv(r'C:\Users\eniad\Downloads\Achievement 6\02 Data\Prepared Data\US_briths.csv')